torch/distributed/algorithms/ddp_comm_hooks/powerSGD_hook.py

```diff
@@ -127,6 +175,20 @@ def powerSGD_hook(state: PowerSGDState, bucket) -> torch.futures.Future:
 
     # The input tensor is a flattened 1D tensor.
     input_tensor = bucket.get_tensors()[0]
+
+    # Run vanilla allreduce in the first `start_powerSGD_iter` iterations.
+    if state.iter < state.start_powerSGD_iter:
+        fut = dist.all_reduce(
+            input_tensor, group=group_to_use, async_op=True
+        ).get_future()
+
+        def div_callback(fut):
+            return [fut.value()[0].div_(world_size)]
+
+        state.maybe_increase_iter(bucket)
+        return fut.then(div_callback)
```

Comment:
Can we directly call into an allreduce hook or some helper that already exists to do the vanilla allreduce? Something like this would be nice:

```
state.maybe_increase(bucket)
return allreduce_fut(...)
```

Reply:
Sure. Let me do that in a follow-up PR.